code/modules/integrated_electronics/subtypes/lists.dm

```diff
@@ -154,25 +172,27 @@
 	var/list/input_list = get_pin_data(IC_INPUT, 1)
 	var/list/red_list = list()
 	var/index = get_pin_data(IC_INPUT, 2)
+	var/key = get_pin_data(IC_INPUT, 3)
 
 	if(length(input_list))
-		for(var/j in 1 to input_list.len)
-			var/I = input_list[j]
-			if(j != index)
-				red_list.Add(I)
+		red_list = input_list.Copy()
+		if(index)
+			red_list.Cut(index, index+1)
+		if(key)
+			red_list -= key
 	set_pin_data(IC_OUTPUT, 1, red_list)
 	push_data()
 	activate_pin(2)
 
-
 /obj/item/integrated_circuit/lists/write
 	name = "write circuit"
-	desc = "This circuit will write an element to a list at the given index location."
+	desc = "This circuit will write an element to a list at the given index location. If value is specified, it will attempt to set the value of the item written to the value. Associations only work if the itme written is a text string."
 	extended_desc = "If there is no element with such index, it will give the same list as before."
 	inputs = list(
 		"list" = IC_PINTYPE_LIST,
 		"index" = IC_PINTYPE_INDEX,
-		"item" = IC_PINTYPE_ANY
+		"item" = IC_PINTYPE_ANY,
```

Comment:
maybe better to make index any?

Reply:
index is the position in the list, not key, if they just want to append something use append circuit, intentional.

Reply:
What if I want to write new value to key?

Reply:
append circuit